src/global/signal_handler.cc

```diff
@@ -90,7 +92,7 @@ static void handle_fatal_signal(int signum)
   // presumably dump core-- will handle it.
   char buf[1024];
   char pthread_name[16] = {0}; //limited by 16B include terminating null byte.
-#if !defined(__FreeBSD__)
+#ifdef HAVE_PTHREAD_GETNAME_NP
   int r = pthread_getname_np(pthread_self(), pthread_name, sizeof(pthread_name));
```

Comment:
again, could have an empty definition which evaluates to "0" for this function if `HAVE_PTHREAD_GETNAME_NP` is not defined.